consensus/consortium/v2/consortium.go

```diff
@@ -462,6 +463,58 @@ func (c *Consortium) verifySeal(chain consensus.ChainHeaderReader, header *types
 	return nil
 }
 
+func (c *Consortium) backOffTime(header *types.Header, snapshot *Snapshot) uint64 {
+	coinbase := header.Coinbase
+	if snapshot.inturn(coinbase) {
+		return 0
+	}
+
+	distance := snapshot.distanceToInturn(coinbase)
+	source := rand.NewSource(header.Number.Int64())
+	rand := rand.New(source)
+
+	// Every other not inturn validator has a different delay.
+	// The delay equals to their random delayMultiplier * wiggleTime.
+	// The delayMultiplier is random in range [1, len(validators)]
+	delayMultiplier := make([]int, 0, len(snapshot.Validators))
+	for i := 0; i < len(snapshot.Validators); i++ {
+		delayMultiplier = append(delayMultiplier, i+1)
+	}
+	rand.Shuffle(len(delayMultiplier), func(i, j int) {
+		delayMultiplier[i], delayMultiplier[j] = delayMultiplier[j], delayMultiplier[i]
+	})
+
+	return uint64(int(wiggleTime) + delayMultiplier[distance]*int(wiggleTime)/2)
```

Comment:
I still think two layers of randomize (rand.Shuffle and inside distanceToInturn)  is over complicated. Can you elaborate?

Reply:
```go
	source := rand.NewSource(header.Number.Int64())
	rand := rand.New(source)
	randomNumber := rand.Intn(len(snapshot.Validators) - 1)
```

Is this not enough?

Reply:
we need a better distribution through every validators delay to minimize block broadcast conflicts, every delay from every node should be different